server/script/migrateAssetsLocation.js

```diff
@@ -24,224 +22,255 @@ const REVISION_TYPES = ['REPOSITORY', 'ACTIVITY', 'CONTENT_ELEMENT'];
 const CHUNK_SIZE = 2000;
 
 const mapEntityToAction = {
-  REPOSITORY: migrateRepository,
-  ACTIVITY: migrateActivity,
-  CONTENT_ELEMENT: migrateContentElement
+  REPOSITORY: 'migrateRepository',
+  ACTIVITY: 'migrateActivity',
+  CONTENT_ELEMENT: 'migrateContentElement'
 };
 
-migrate()
-  .then(({ transaction }) => transaction.commit())
-  .then(() => {
-    console.info('Migration script was executed successfully.');
-    process.exit(0);
-  })
-  .catch(error => {
-    console.error(error.message);
-    process.exit(1);
-  });
-
-async function migrate() {
-  const transaction = await sequelize.transaction();
-  const tasks = await getMigrationTasks(transaction);
-  return tasks.run({ transaction });
-}
+class Migration {
+  async initialize() {
+    this.transaction = await sequelize.transaction();
+    this.schemasMeta = getFileMetas(SCHEMAS);
+  }
 
-async function getMigrationTasks(transaction) {
-  const repositories = await Repository.findAll({ transaction });
-  const tasks = repositories.map(repository => ({
-    title: `Migrate repository: "${repository.name}"`,
-    task: () => getRepositoryTasks(repository, transaction)
-  }));
-  return new Listr(tasks);
-}
+  async getTasks() {
+    const { schemasMeta, transaction } = this;
+    const repositories = await Repository.findAll({ transaction });
+    const tasks = repositories.map(repository => ({
+      title: `Migrate repository "${repository.name}"`,
+      task: () => {
+        const schemaMeta = get(schemasMeta, repository.schema);
+        const repositoryMigration = new RepositoryMigration({ repository, schemaMeta, transaction });
+        return repositoryMigration.getTasks();
+      }
+    }));
+    return new Listr(tasks);
+  }
 
-async function getRepositoryTasks(repository, transaction) {
-  const schemaFileMetas = getFileMetasForSchema(repository.schema);
-  return new Listr([
-    {
-      title: 'Migrate repository',
-      task: () => migrateRepositoryAssets(repository, schemaFileMetas, transaction)
-    },
-    {
-      title: 'Migrate activities',
-      task: () => migrateRepositoryActivities(repository.id, schemaFileMetas, transaction)
-    },
-    {
-      title: 'Migrate content elements',
-      task: () => migrateRepositoryContentElements(repository.id, schemaFileMetas, transaction)
-    },
-    {
-      title: 'Migrate revisions',
-      task: () => migrateRepositoryRevisions(repository.id, schemaFileMetas, transaction)
-    }
-  ]);
+  async run() {
+    const tasks = await this.getTasks();
+    return tasks.run().then(() => this.transaction.commit());
+  }
```

Comment:
I'm thinking to move this Migration class logic outside the class and put it inside `migrate` function and just keep `RepositoryMigration` class. Doing it that way I will avoid this: 
```js
migration.initialize().then(() => {
  migration.run()
    .then(() => {
      console.info('Migration script was executed successfully.');
      process.exit(0);
    })
    .catch(error => {
      console.error(error.message);
      process.exit(1);
    });
});
```
where I have nested `.then` methods.
@underscope what do you think?